clients/src/test/java/org/apache/kafka/clients/telemetry/TelemetryManagementInterfaceTest.java

```diff
@@ -102,47 +100,208 @@ public void testValidateClientInstanceId() {
+
+        testValidateTransition(currState, validStates);
+    }
+
+    @Test
+    public void testValidateTransitionForTerminated() {
+        TelemetryState currState = TelemetryState.terminated;
+
+        // There's no transitioning out of the terminated state
+        testValidateTransition(currState, Collections.emptyList());
+    }
+
+    private void testValidateTransition(TelemetryState oldState, List<TelemetryState> validStates) {
+        for (TelemetryState newState : validStates)
+            TelemetryManagementInterface.validateTransition(oldState, newState);
+
+        // Have to copy to a new list because asList returns an unmodifiable list
+        List<TelemetryState> invalidStates = new ArrayList<>(Arrays.asList(TelemetryState.values()));
+
+        // Remove the valid states from the list of all states, leaving only the invalid
+        invalidStates.removeAll(validStates);
+
+        for (TelemetryState newState : invalidStates) {
+            Executable e = () -> TelemetryManagementInterface.validateTransition(oldState, newState);
+            String unexpectedSuccessMessage = "Should have thrown an IllegalTelemetryStateException for transitioning from " + oldState + " to " + newState;
+            assertThrows(IllegalTelemetryStateException.class, e, unexpectedSuccessMessage);
         }
+    }
 
-        String s = new String(Utils.readBytes((ByteBuffer) decompressed.flip()));
-        String expected = String.format("%s: %s\n%s: %s\n", telemetryMetric1.name(), telemetryMetric1.value(), telemetryMetric2.name(), telemetryMetric2.value());
-        assertEquals(expected, s);
+    private void testMaybeCreateFailsIfParametersAreNull(boolean enableMetricsPush, Time time, String clientId) {
+        // maybeCreate won't (or at least it shouldn't) fail if these are both non-null
+        if (time != null && clientId != null)
+            return;
+
+        // maybeCreate won't fail if we don't attempt to construct metrics in the first place
+        if (!enableMetricsPush)
+            return;
+
+        Class<IllegalArgumentException> e = IllegalArgumentException.class;
+        ProducerConfig config = newConfig();
+
+        assertThrows(e,
+            () -> TelemetryManagementInterface.maybeCreate(true, time, clientId),
+            String.format("maybeCreate should have thrown a %s for time: %s and clientId: %s", e.getName(), time, clientId));
+
+        assertThrows(e,
+            () -> TelemetryManagementInterface.maybeCreate(config, time, clientId),
+            String.format("maybeCreate should have thrown a %s for time: %s, clientId: %s",  e.getName(), time, clientId));
     }
 
-    private TelemetryMetric telemetryMetric(String name, long value) {
+    private TelemetryMetric newTelemetryMetric(String name, long value) {
         return new TelemetryMetric(name,
             MetricType.sum,
             value,
             "Description for " + name);
     }
+
+    private ProducerConfig newConfig() {
```

Comment:
maybe something like bootstrapTestProducerConfig?

Reply:
Changed to suggested name.